### PR TITLE
SYAL-1242 update filtering and status device

### PR DESCRIPTION
**Pull request by HieuLM-AVISPL**
Hi @MaksimR-AVISPL. This is the code of the Google Workspace adapter after updating filter, status device. Please help me to review it. Thanks

### CODE REVIEW

File: src/main/java/com/avispl/symphony/dal/communicator/aggregator/common/GoogleWorkspaceConstant.java [hunk 1]
```diff
@@ -64,4 +65,7 @@ public class GoogleWorkspaceConstant {
 	public static final String HOUR = " h ";
 	public static final int MAC_ADDRESS_LENGTH = 12;
 	public static final int DEFAULT_ORG_UNIT_POSITION = 0;
+	public static final int MIN_VOLUME_LEVEL = 0;
+	public static final String REGEX_SERIAL_NUMBER = "^[a-zA-Z0-9]+$";
+	public static final String CPU_TEMPERATURE = "Core1(C)";
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/communicator/aggregator/common/GoogleWorkspaceConstant.java]: Is it always gonna be "Core1"? Wonder if we need 1 then

File: src/main/java/com/avispl/symphony/dal/communicator/aggregator/GoogleWorkspaceCommunicator.java [hunk 2]
```diff
@@ -920,25 +953,23 @@ private void mapMonitoringProperty(Map<String, String> mappingStatistic, Map<Str
...
-				List<CPUTemperature> cpuTemperatures = convertJsonNodeToList(lastObject, new TypeReference<List<CPUTemperature>>() {
-				});
-				if (!cpuTemperatures.isEmpty()) {
-					String label = CPUTemperatureEnum.getByName(name).getValue();
-					int temperature = cpuTemperatures.stream().filter(cpuTemperature -> cpuTemperature.getLabel().equals(label)).map(CPUTemperature::getTemperature).findFirst()
-							.orElse(0);
+		JsonNode jsonNodeValue = getJsonNodeValue(value);
 
+		if (jsonNodeValue != null) {
+			JsonNode lastObject = jsonNodeValue.get(jsonNodeValue.size() - 1).get(GoogleWorkspaceConstant.CPU_TEMPERATURE_INFO);
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/communicator/aggregator/GoogleWorkspaceCommunicator.java:962]: is it safe to do jsonNodeValue.size() - 1 here? 
the blocks only checks for the null value, is it possible that there's an empty array, so the result ends up being negative?

File: src/main/java/com/avispl/symphony/dal/communicator/aggregator/GoogleWorkspaceCommunicator.java [hunk 3]
```diff
@@ -1037,9 +1086,19 @@ private JsonNode getJsonNodeByDeviceId(JsonNode jsonNode, String id) {
...
 			return filterOrgUnit;
 		}
+		if (StringUtils.isNotNullOrEmpty(filterSerialNumber) && aggregatedDeviceResponse.size() > 0) {
+			for (JsonNode item : aggregatedDeviceResponse) {
+				orgUnitNameList.add(item.get(GoogleWorkspaceConstant.ORG_UNIT_PATH).asText().substring(1));
+			}
+			if (StringUtils.isNotNullOrEmpty(currentOrgUnitName)) {
+				return currentOrgUnitName;
+			}
+			return aggregatedDeviceResponse.get(GoogleWorkspaceConstant.DEFAULT_ORG_UNIT_POSITION).get(GoogleWorkspaceConstant.ORG_UNIT_PATH).asText().substring(1);
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/communicator/aggregator/GoogleWorkspaceCommunicator.java]: get(GoogleWorkspaceConstant.DEFAULT_ORG_UNIT_POSITION) and get(GoogleWorkspaceConstant.ORG_UNIT_PATH) are always gonna be here, right? Just worried about potential NPEs

**Review by MaksimR-AVISPL — APPROVED** [review]

File: src/main/java/com/avispl/symphony/dal/communicator/aggregator/GoogleWorkspaceCommunicator.java (hunk 2 above)
Comment (HieuLM-AVISPL) [reply src/main/java/com/avispl/symphony/dal/communicator/aggregator/GoogleWorkspaceCommunicator.java:962]: > is it safe to do jsonNodeValue.size() - 1 here? the blocks only checks for the null value, is it possible that there's an empty array, so the result ends up being negative?

I added a condition to it

File: src/main/java/com/avispl/symphony/dal/communicator/aggregator/common/GoogleWorkspaceConstant.java (hunk 1 above)
Comment (HieuLM-AVISPL) [reply src/main/java/com/avispl/symphony/dal/communicator/aggregator/common/GoogleWorkspaceConstant.java]: > Is it always gonna be "Core1"? Wonder if we need 1 then

I renamed it "CPU Temperature", it includes data of all Core

**Pull request merged**